### PR TITLE
Kev/490_remain

### PR DESCRIPTION
**Pull request by junhaow1**
# Pull Request Details

## What issue does this PR address

- NAVIGATION: Remain on current page after a BUILD except if on page 0 #490

- Link to associated issue: #490 

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [ ] Pre-exisiting lint errors noted: [HERE]
- [ ] Integration test `make qtest.tmp` screenshot included in issue
- [ ] Tested on device:
  - [ ] Linux
  - [x] MacOS
  - [ ] Windows
- [ ] Added two reviewers

## Finalising

Once PR discussion is complete and reviewers have approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev (gjwgit)

### CODE REVIEW

**Comment by junhaow1** [comment]
image

File: lib/widgets/activity_button.dart
Comment (gjwgit) [inline lib/widgets/activity_button.dart:1]: Why not call it onPressed rather than additionalLogic?

File: lib/utils/page_navigation_logic.dart
Comment (gjwgit) [inline lib/utils/page_navigation_logic.dart:1]: Where is this used?

**Review by gjwgit — CHANGES_REQUESTED** [review]
Need a walk through on this to clarify the thinking.

File: lib/widgets/activity_button.dart [hunk 1]
```diff
@@ -65,7 +67,28 @@ class ActivityButton extends ConsumerWidget {
...
+          if (pageControllerProvider != null) {
+            // Access the PageController directly from the StateProvider.
+            final pageController = ref.read(pageControllerProvider!);
+
+            // Check the current page index before navigating.
+            final currentPage = pageController.page?.round() ?? 0;
+
+            // Determine the target page index based on the current page.
+            int targetPage =
+                (currentPage >= 2 && currentPage <= 4) ? currentPage : 1;
```
Comment (gjwgit) [inline lib/widgets/activity_button.dart]: Why less than 4?

**Review by gjwgit — APPROVED** [review]

**Pull request merged**